#The sequential Model

In [ ]:
import tensorflow as tf
import keras
from keras import layers

A Sequential model is appropriate for a plain stack of layers where each layer has exactly one input tensor and one output tensor.

**Dense  Layer -**
This layer is widely used in Keras for creating the deeply connected layer in the neural network

**Flatten Layer**
This layer is used in input flattening

**Dropout Layer**
This layer is used to reduce the overfitting of the neural network.


In [ ]:
model = keras.Sequential([
    layers.Dense(3 , activation = 'relu' ,name ='Inputlayer'),
     layers.Dense(4,activation = 'relu',name = 'Hiddenlayer'),
     layers.Dense(4, name = 'output layer')

                         ])

x= tf.ones((2,2))
model(x)

<tf.Tensor: shape=(2, 4), dtype=float32, numpy=
array([[-0.10146073,  0.13315804,  0.04760575,  0.12365723],
       [-0.10146073,  0.13315804,  0.04760575,  0.12365723]],
      dtype=float32)>

In [ ]:
layer1 = layers.Dense(3,activation = 'relu', name = 'InputLayer')
layer2 = layers.Dense(4,activation = 'relu', name = 'HiddenLayer')
layer3 = layers.Dense(4 , name = 'OutputLayer')

x = tf.ones((2,2))
y = layer3(layer2(layer1(x)))
y

<tf.Tensor: shape=(2, 4), dtype=float32, numpy=
array([[0., 0., 0., 0.],
       [0., 0., 0., 0.]], dtype=float32)>

**Once** a model is "built", you can call its summary() method to display its contents:

In [ ]:
model.summary()

Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 Inputlayer (Dense)          (2, 3)                    9         
                                                                 
 Hiddenlayer (Dense)         (2, 4)                    16        
                                                                 
 output layer (Dense)        multiple                  20        
                                                                 
Total params: 45 (180.00 Byte)
Trainable params: 45 (180.00 Byte)
Non-trainable params: 0 (0.00 Byte)
_________________________________________________________________


In [ ]:
from keras.models import Sequential
from keras.layers import Flatten,Dense

model = Sequential()
model.add( Flatten(input_shape = (28,28) ))
model.add(Dense(512, activation = 'relu'))
model.add(Dense(512, activation = 'relu'))
model.add(Dense(10, activation = 'softmax'))
model.summary()



Model: "sequential_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 flatten (Flatten)           (None, 784)               0         
                                                                 
 dense (Dense)               (None, 512)               401920    
                                                                 
 dense_1 (Dense)             (None, 512)               262656    
                                                                 
 dense_2 (Dense)             (None, 10)                5130      
                                                                 
Total params: 669706 (2.55 MB)
Trainable params: 669706 (2.55 MB)
Non-trainable params: 0 (0.00 Byte)
_________________________________________________________________


Params after the flatten layer = 0, because this layer only flattens the image to a vector for feeding into the input layer. The weights haven’t been added yet.

Params after layer 1 = (784 nodes in input layer) × (512 in hidden layer 1) + (512 connections to biases) = 401,920.

Params after layer 2 = (512 nodes in hidden layer 1) × (512 in hidden layer 2) + (512 connections to biases) = 262,656.

Params after layer 3= (512 nodes in hidden layer 2) × (10 in output layer) + (10 connections to biases) = 5,130.

Total params in the network = 401,920 + 262,656 + 5,130 = 669,706.
